tools/src/main/java/org/apache/kafka/tools/streams/StreamsGroupCommand.java

```diff
@@ -156,6 +167,158 @@ private void printGroupInfo(List<GroupListing> groups) {
+            return descriptionMap.get(group);
+        }
+
+        private void printMembers(StreamsGroupDescription description, boolean verbose) {
+            int groupLen = Math.max(15, description.groupId().length());
+            int maxMemberIdLen = 15, maxHostLen = 15, maxClientIdLen = 15;
+            Collection<StreamsGroupMemberDescription> members = description.members();
+            if (maybePrintEmptyGroupState(description.groupId(), description.groupState(), description.members().size())) {
+                for (StreamsGroupMemberDescription member : members) {
+                    maxMemberIdLen = Math.max(maxMemberIdLen, member.memberId().length());
+                    maxHostLen = Math.max(maxHostLen, member.processId().length());
+                    maxClientIdLen = Math.max(maxClientIdLen, member.clientId().length());
+                }
+
+                if (!verbose) {
+                    String fmt = "%" + -groupLen + "s %" + -maxMemberIdLen + "s %" + -maxHostLen + "s %" + -maxClientIdLen + "s\n"
+                        + "%s %s %s\n\n";
+                    System.out.printf(fmt, "GROUP", "MEMBER", "PROCESS", "CLIENT-ID", "ACTIVE-TASKS", "STANDBY-TASKS", "WARMUP-TASKS");
+                    for (StreamsGroupMemberDescription member : members) {
+                        System.out.printf(fmt, description.groupId(), member.memberId(), member.processId(), member.clientId(),
+                            getTopicPartitions(member.assignment().activeTasks()).stream().map(tp -> tp.topic() + ":" + tp.partition()).collect(Collectors.joining(",")),
+                            getTopicPartitions(member.assignment().standbyTasks()).stream().map(tp -> tp.topic() + ":" + tp.partition()).collect(Collectors.joining(",")),
+                            getTopicPartitions(member.assignment().warmupTasks()).stream().map(tp -> tp.topic() + ":" + tp.partition()).collect(Collectors.joining(",")));
+                    }
+                } else {
+                    String fmt = "%" + -groupLen + "s %-15s%" + -maxMemberIdLen + "s %s %15s %" + -maxHostLen + "s %" + -maxClientIdLen + "s\n"
+                        + "%s %s %s %s %s %s\n\n";
+                    System.out.printf(fmt, "GROUP", "TARGET-ASSIGNMENT-EPOCH", "MEMBER", "MEMBER-PROTOCOL", "MEMBER-EPOCH", "PROCESS", "CLIENT",
+                        "ACTIVE-TASKS", "STANDBY-TASKS", "WARMUP-TASKS", "TARGET-ACTIVE-TASKS", "TARGET-STANDBY-TASKS", "TARGET-WARMUP-TASKS");
+                    for (StreamsGroupMemberDescription member : members) {
+                        System.out.printf(fmt, description.groupId(), description.targetAssignmentEpoch(), member.memberId(), member.isClassic() ? "Classic Protocol" : "New Protocol", member.memberEpoch(), member.processId(), member.clientId(),
+                            getTopicPartitions(member.assignment().activeTasks()).stream().map(tp -> tp.topic() + ":" + tp.partition()).collect(Collectors.joining(",")),
```

Comment:
This does not make much sense to me. Why are converting tasks to topic partitions to print them?

Maybe just implement a method taskIdsToString and use it everywhere.